src/panel/direction/MobileRoadMapPreview.jsx

```diff
@@ -2,62 +2,79 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import RoadMapStep from './RoadMapStep';
 import { fire } from 'src/libs/customEvents';
-import { CloseButton } from 'src/components/ui';
 
 export default class MobileRoadMapPreview extends React.Component {
   static propTypes = {
     steps: PropTypes.array.isRequired,
     onClose: PropTypes.func.isRequired,
-  }
+  };
 
-  state = {
-    currentStep: 0,
-  }
+  state = { };
 
   componentDidMount() {
-    this.zoomToCurrentStep();
-    fire('move_mobile_geolocation_button', 65);
+    this.steps = document.querySelector('.mobile-roadmap-preview-steps');
+    this.stepList = document.querySelectorAll('.itinerary_mobile_step');
+    this.currentStep = 0;
```

Comment:
… but actually, this is typically something that should be stored in the state :)
This perfectly matches the definition of it: an internal value storing a part of the current component state which, when it changes, should result in a visual change.

Reply:
Hi, thanks for the review, I actually started by doing this stuff in the proper "react" way, but when I did that, I had the impression that the snap-scroll seemed more janky, or buggy when the setState happened in the middle of a scroll. Maybe it was just a problem with my PC or my Chrome's mobile view, we could put the setStates back and verify it on real mobile devices if you want ...?

Reply:
Hi!
That would be great to restore, yes :)

About the perf, there are common problems, not related to React in itself, when you plug DOM modifications directly to scroll events, as these are generated very quickly and can cause the DOM to "choke" with too many changes.

But I think if you keep you test in the scroll method to avoid doing a `this.setState({ currentStep })` when it's not needed, it should be fine. And if not we can still using throttling strategies but probably it won't be needed here.